### PR TITLE
Eng 11296 code cov

### PR DESCRIPTION
**Pull request by pdshaw**
Remove EMMA code coverage references and JAR's
Enable JaCoCo agent in build if environment variable USE_JACOCO set
Leave everything that used to work still working.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -685,7 +686,7 @@ DISTRIBUTION
             <include name="src/frontend/org/voltdb/VoltProcedure.java" />
             <include name="src/frontend/org/voltdb/SQLStmt.java" />
             <include name="src/frontend/org/voltdb/VoltType.java" />
-            <include name="src/frontend/org/voltdb/InProcessVoltDBServer.java" />
+            <include name="src/frontend/org/voltdb/ProcInfo.java" />
```
Comment (brians127) [inline build.xml]: What does this change do and why is it needed?

File: build.xml [hunk 2]
```diff
@@ -1944,7 +1945,6 @@ TEST CASES
         <!-- Fail the build if there were any problems.
              This runs all the tests before failing. -->
         <available file="${build.testoutput.dir}/JUNITHADFAILURES" property="junit.failures"/>
-        <fail if="junit.failures" unless="emma.enabled" message="JUnit had failures" />
```
Comment (brians127) [inline build.xml]: Shouldn't this be "fail if junit.failures unless jacoco.enabled"?

Comment (pdshaw) [inline build.xml]: Yes essentially. Re-added with check for use_jacoco.

Comment (brians127) [inline build.xml]: Thanks! My concerns have been addressed.

File: build.xml [hunk 3]
```diff
@@ -26,6 +26,7 @@
             includes="@{includes}"
             encoding='UTF-8'
             debug='true'
+            debuglevel="lines"
```
Comment (nshi) [inline build.xml]: Does this increase the kit jar size by including line level debug info?

Comment (pdshaw) [inline build.xml]: Good catch - I'll make it conditional on the use_jacoco setting.

Growth is 750k bytes, up from 211400017 without lines.

File: build.xml [hunk 4]
```diff
@@ -1686,6 +1689,7 @@ TEST CASES
             <!-- <jvmarg value="-Xcheck:jni" /> -->
             <jvmarg value="-Xmx256m"/>
             <jvmarg value="-XX:+HeapDumpOnOutOfMemoryError"/>
+            <jvmarg if:set="use_jacoco" value="-javaagent:third_party/java/jars/jacocoagent.jar"/>
```
Comment (akhanzode) [inline build.xml]: use vendor.lib.dir also create a property for jarname so we can easily upgrade change.

File: build.xml [hunk 5]
```diff
@@ -1991,64 +1992,39 @@ TEST CASES
...
-        depends="with.emma, junit, emma-report"
-/>
-
-<target name='emma_junit_legacy'
-        description="run junit_legacy target and generate coverage report"
-        depends="with.emma, junit_legacy, emma-report"
-/>
+<!-- Resources for the JaCoCo task below. -->
+<taskdef uri="antlib:org.jacoco.ant" resource="org/jacoco/ant/antlib.xml">
+	<classpath path="third_party/java/jars/jacocoant.jar"/>
```
Comment (akhanzode) [inline build.xml]: use property.

File: build.xml [hunk 6]
```diff
@@ -1991,64 +1992,39 @@ TEST CASES
...
-            <xml outfile="${emma.dir}/coverage.xml" depth="method" />
-            <html
-                outfile="${emma.dir}/coverage.html"
-                depth="method"
-                columns="name,class,method,block,line"
-                encoding="UTF-8"
-            />
-        </report>
-    </emma>
+        <html destdir="report"/>
```
Comment (akhanzode) [inline build.xml]: make dirname expllicit like jacoco-report

**Comment by akhanzode** [comment]
Update README.thirdparty check with Ruth if its needed.

File: build.xml (hunk 6 above)
Comment (pdshaw) [inline build.xml]: done

File: build.xml (hunk 5 above)
Comment (pdshaw) [inline build.xml]: done

File: build.xml (hunk 4 above)
Comment (pdshaw) [inline build.xml]: done

**Pull request merged**